src/azure-cli/azure/cli/command_modules/acr/tests/latest/test_acr_agentpool_commands.py

```diff
@@ -0,0 +1,85 @@
+                         self.check('count', 1),
+                         self.check('tier', 'S1'),
+                         self.check('virtualNetworkSubnetResourceId', 'None'),
+                         self.check('provisioningState', 'Succeeded'),
+                         self.check('os', 'Linux'),
+                         self.check('location', '{rg_loc}'),
+                         self.check('resourceGroup', '{rg}')])
+
+        # Create a S2 tier agentpool in a VNET.
+        response = self.cmd('network vnet create -n {vnet_name} --subnet-name {subnet_name} -g {rg} -l {rg_loc}',
+                            checks=[self.check('newVNet.name', '{vnet_name}'),
+                                    self.check('newVNet.location', '{rg_loc}'),
+                                    self.check('newVNet.provisioningState', 'Succeeded'),
+                                    self.check('newVNet.subnets[0].name', '{subnet_name}'),
+                                    self.check('newVNet.subnets[0].provisioningState', 'Succeeded')]).get_output_in_json()
+        self.kwargs.update({
+            'subnet_id': response['newVNet']['subnets'][0]['id']
+        })
+        self.cmd('acr agentpool create -n {agents2_name} -r {registry_name} --tier s2 --subnet-id {subnet_id}',
+                 checks=[self.check('name', '{agents2_name}'),
+                         self.check('count', 1),
+                         self.check('tier', 'S2'),
+                         self.check('virtualNetworkSubnetResourceId', '{subnet_id}'),
+                         self.check('provisioningState', 'Succeeded'),
+                         self.check('os', 'Linux'),
+                         self.check('location', '{rg_loc}'),
+                         self.check('resourceGroup', '{rg}')])
+
+        # list agentpools
+        self.cmd('acr agentpool list -r {registry_name}',
+                 checks=[self.check('[0].name', '{agents2_name}'),
+                         self.check('[1].name', '{agents1_name}')])
+
+        # show agentpool properties
+        self.cmd('acr agentpool show -n {agents1_name} -r {registry_name}',
+                 checks=[self.check('name', '{agents1_name}')])
+
+        # update the first agentpool using non-default parameter values
+        self.cmd('acr agentpool update -n {agents1_name} -r {registry_name} -c 2',
+                 checks=[self.check('name', '{agents1_name}'),
+                         self.check('count', 2),
+                         self.check('tier', 'S1'),
+                         self.check('virtualNetworkSubnetResourceId', 'None'),
+                         self.check('provisioningState', 'Succeeded'),
+                         self.check('os', 'Linux'),
+                         self.check('location', '{rg_loc}'),
+                         self.check('resourceGroup', '{rg}')])
+
+        # test agentpool delete
+        #self.cmd('acr agentpool delete -n {agents2_name} -r {registry_name} -y')
```

Comment:
Is this command not working?

Reply:
it is, but I have some polling built in the command to handle a scenario that it's not handle by the cli polling and the testing gives me back an error. I can't shake it off. I'm trying a lot of stuff to do it and I wanted to make sure it was that command because locally all the testing passes and it is that command.